### ISSUE TITLE
Question about surc

### ISSUE THREAD

**Issue by pumpingiron** [amaya382/anyrc#17]
Hello,

I use those kind of functions for quite some time now, because I already used it when its was **Russell91/sshrc**, and as a consequence and as quoted in reference, I'd like to make a surc as root for example and preserve my user's environment variables in order to keep my powerline prompt and other stuff like that working without having to install it under root for example.

It was possible with sshrc by writing a .sshrc file which must be the equivalent of your .anyrc file like the following :

**# Define surc function to keep sshrc settings when su

surc () {
   [ "$1" = "-" ] && shift
   local USER="$1"
   if [ -z "${USER}" ]; then
     USER="root"
     chmod +x "${SSHHOME}" 2>/dev/null
     sudo -u "${USER}" SSHHOME="${SSHHOME}" "${SSHHOME}/bashsshrc"
   fi
}**

It was the method suggested at that time, so my question now is "have you got an equivalent already present somewhere that i haven"t seen, or would it be possible to integrate it according to you ?

Thank you in advance for your answer.

**Comment by amaya382** [comment]
Hi @pumpingiron ,

You can simply use `surc`, which is a part of this project, by creating a proper .anyrc. It should work the same as `surc` you defined.
If you want to preserve env vars in addition, pass `-E` flag provided by `sudo` to `surc`. Currently, `surc` uses `sudo` internally.

Here is a pseudo example
```console
$ whoami
someuser
$ cat .anyrc
bash() {
  $(sh -c 'which bash') --rcfile "${ANYRC_HOME}/.bashrc"
}
$ cat .anyrc.d/.bashrc
alias ll='ls -Al'
source "${ANYRC_DANYRC}"
$ source .anyrc.d/.bashrc
$ ll # of course, you can use `ll` defined in .bashrc
$ export FOO=foo
$ surc -E root
# whoami
root
# echo $FOO
foo
# ll # you can also use `ll` defined in .bashrc w/ any accounts' env
# exit
$ whoami
someuser
```

**Comment by pumpingiron** [comment]
Hi @amaya382 ,

To tell you everything, I had already looked there, but my container got into trouble because I had several problems in my environment's custom configuration, so I prefered asking you how you would do it to save time.

Your reflection let me identify where the problems where and I was on the right track, and finally my configuration was well designed for historical **Russell91/sshrc** but not for yours because you really well improved it with all the mechanics that make the surc I've provided in my previous post unnecessary, and after having upgraded my configuration to take into account all of you improvements, i'm now able again to surc with what is expected.

Big thanks.
